web/src/timestamp_width_handler.js

```diff
@@ -0,0 +1,231 @@
+import * as timerender from "./timerender";
+import {user_settings} from "./user_settings";
+
+// candidate_timestamps are contains max possible timestamps in every locale
+const candidate_timestamps = {
```

Comment:
Okay, again, this is not the right approach. You'd want a couple of Date-object candidates with times for, say, 11:59 AM (11:59) and 11:59 PM (23:59). We don't want to have to maintain this massive list of approximations of what the localized string will be.

Reply:
'You'd want a couple of Date-object candidates with times for, say, 11:59 AM (11:59) and 11:59 PM (23:59).' if we  only take some Date-object candidates, it will cause an issue because in every locale the timestamp size changes, also if your idea is of calculating candidate timestamps every time when the time is rendered I think it would take a lot of time and it will have to do calculations every time.

Reply:
I can change the timestamps to possible date objects rather than using strings "11:59 AM", and then use the function `timerender.stringify_time()` to convert the date object into the time which was being rendered previously.